lib/ruby_smb/generic_packet.rb

```diff
@@ -42,7 +42,7 @@ def packet_smb_version
     def self.read(val)
       begin
         super(val)
-      rescue EOFError => e
+      rescue EOFError, IOError => e
```

Comment:
If I remember well, EOFError is a descendant of the IOError class. So, I think using `rescue IOError => e` should be enough.